File: GeneratedEntityReaderClasses/Components/MaterialSuckerComponent.cs
namespace GeneratedNoitaClasses;
public class MaterialSuckerComponent : NoitaComponentBase,  iNoitaType<MaterialSuckerComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaInt material_type { get; set; } = new();
	public NoitaInt barrel_size { get; set; } = new();
	public NoitaInt num_cells_sucked_per_frame { get; set; } = new();
	public NoitaBool set_projectile_to_liquid { get; set; } = new();
	public NoitaInt last_material_id { get; set; } = new();
	public NoitaBool suck_gold { get; set; } = new();
	public NoitaBool suck_health { get; set; } = new();
	public NoitaBool suck_static_materials { get; set; } = new();
	public NoitaString suck_tag { get; set; } = new();
	public types_iaabb randomized_position { get; set; } = new();
	public NoitaInt mAmountUsed { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		material_type.Read(s);
		barrel_size.Read(s);
		num_cells_sucked_per_frame.Read(s);
		set_projectile_to_liquid.Read(s);
		last_material_id.Read(s);
		suck_gold.Read(s);
		suck_health.Read(s);
		suck_static_materials.Read(s);
		suck_tag.Read(s);
		randomized_position.Read(s);
		mAmountUsed.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		material_type.Write(s);
		barrel_size.Write(s);
		num_cells_sucked_per_frame.Write(s);
		set_projectile_to_liquid.Write(s);
		last_material_id.Write(s);
		suck_gold.Write(s);
		suck_health.Write(s);
		suck_static_materials.Write(s);
		suck_tag.Write(s);
		randomized_position.Write(s);
		mAmountUsed.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/MaterialSuckerComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaInt
from GeneratedClasses.types_iaabb import types_iaabb

from NoitaStream import NoitaStream
@dataclass
class MaterialSuckerComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    material_type:NoitaInt = field(default_factory=NoitaInt)
    barrel_size:NoitaInt = field(default_factory=NoitaInt)
    num_cells_sucked_per_frame:NoitaInt = field(default_factory=NoitaInt)
    set_projectile_to_liquid:NoitaBool = field(default_factory=NoitaBool)
    last_material_id:NoitaInt = field(default_factory=NoitaInt)
    suck_gold:NoitaBool = field(default_factory=NoitaBool)
    suck_health:NoitaBool = field(default_factory=NoitaBool)
    suck_static_materials:NoitaBool = field(default_factory=NoitaBool)
    suck_tag:NoitaString = field(default_factory=NoitaString)
    randomized_position:types_iaabb = field(default_factory=types_iaabb)
    mAmountUsed:NoitaInt = field(default_factory=NoitaInt)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.material_type.Read(s)
        self.barrel_size.Read(s)
        self.num_cells_sucked_per_frame.Read(s)
        self.set_projectile_to_liquid.Read(s)
        self.last_material_id.Read(s)
        self.suck_gold.Read(s)
        self.suck_health.Read(s)
        self.suck_static_materials.Read(s)
        self.suck_tag.Read(s)
        self.randomized_position.Read(s)
        self.mAmountUsed.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.material_type.Write(s)
        self.barrel_size.Write(s)
        self.num_cells_sucked_per_frame.Write(s)
        self.set_projectile_to_liquid.Write(s)
        self.last_material_id.Write(s)
        self.suck_gold.Write(s)
        self.suck_health.Write(s)
        self.suck_static_materials.Write(s)
        self.suck_tag.Write(s)
        self.randomized_position.Write(s)
        self.mAmountUsed.Write(s)

